### ISSUE TITLE
`packing: true` drops SFT's assistant-only loss mask: TRL 0.29 packs only `input_ids`, so prompt and system tokens are trained (344 of 352 supervised, not 64)

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#1236]
## What happens

SFT stores its loss mask in a `labels` column: -100 on system/user tokens, token ids on the assistant reply. The mask comes from `data.train_on_responses_only`, which defaults to `true`, or from `train_on_messages_with_train_field`. With `training.packing: true`, that column does not survive TRL's packing step, and every packed token becomes a training target.

Reproduced on CPU with the real `SFTTrainerWrapper.setup()` (the fixture follows `tests/test_issue691_sft_packing.py`: a tiny random-init Llama built from a config and a word-level tokenizer, no download). The config uses 8 identical rows of system prompt, user prompt and a short assistant reply, with `data.chat_template: chatml`, `max_length: 128`, `batch_size: 2`, and `packing` set to `false`, then `true`. After `setup()`, the real train dataloader was drained, counting `labels != -100`:

```python
w = SFTTrainerWrapper(cfg, device="cpu")
w.setup({"train": [row] * n_rows})
ds = w.trainer.train_dataset
print(f"[{label}] trainer.train_dataset rows={len(ds)} columns={ds.column_names}")
real = supervised = 0
for batch in w.trainer.get_train_dataloader():
    lab = batch["labels"]
    supervised += int((lab != -100).sum())
    if "attention_mask" in batch:
        real += int(batch["attention_mask"].sum())
    else:
        real += int(batch["input_ids"].numel())
```

```
[responses_only_packing_false] data.train_on_responses_only=True train_on_messages_with_train_field=False training.packing=False
[responses_only_packing_false] trainer.train_dataset rows=8 columns=['input_ids', 'labels', 'attention_mask']
[responses_only_packing_false] dataloader: real tokens=352 supervised label positions=64
[responses_only_packing_true] data.train_on_responses_only=True train_on_messages_with_train_field=False training.packing=True
[responses_only_packing_true] trainer.train_dataset rows=4 columns=['input_ids', 'seq_lengths']
[responses_only_packing_true] dataloader: real tokens=352 supervised label positions=344
[train_field_packing_false] data.train_on_responses_only=False train_on_messages_with_train_field=True training.packing=False
[train_field_packing_false] trainer.train_dataset rows=8 columns=['input_ids', 'labels', 'attention_mask']
[train_field_packing_false] dataloader: real tokens=352 supervised label positions=64
[train_field_packing_true] data.train_on_responses_only=False train_on_messages_with_train_field=True training.packing=True
[train_field_packing_true] trainer.train_dataset rows=4 columns=['input_ids', 'seq_lengths']
[train_field_packing_true] dataloader: real tokens=352 supervised label positions=344
```

With packing on, the `labels` column is gone and 344 of 352 tokens are supervised. The 8 exceptions are the first token of each packed sequence. With packing off, 64 tokens are supervised, the assistant replies only. Nothing warns. TRL's own packing warnings (about attention implementations) do not mention the mask.

## Why

`src/soup_cli/trainer/sft.py:137-166` (`_map_text_sft_rows`) produces `{input_ids, labels, attention_mask}` rows, and `sft.py:1025-1027` passes `packing` into the `SFTConfig` (`_as_sft_config`, `sft.py:1481-1506`). In TRL 0.29.1, `SFTTrainer._prepare_dataset` with packing keeps only `["input_ids"]`, plus `completion_mask` / `assistant_masks` if present (`trl/trainer/sft_trainer.py:1159-1165`). It then packs, so `labels` is dropped. `DataCollatorForLanguageModeling.torch_call` then falls back to `labels = input_ids` (`sft_trainer.py:167-170`) and masks only position 0 of each packed sequence (`:213`). No validator in `config/schema.py` pairs `packing` with a masking mode. The #691 tests assert that rows are packed, never that the mask survives.

## Fix path

**Wire it.** Emit the mask in a column TRL's packing keeps. `assistant_masks = [int(l != -100) for l in labels]` is honoured by TRL's collator unconditionally (`labels[assistant_masks == 0] = -100`) and is concatenated alongside `input_ids` when packing. Soup's `labels` are unshifted, like TRL's, so the positions line up. Keeping `labels` for the unpacked path is fine; the new column only has to exist whenever `packing` is on.

**Or refuse** `packing: true` at config parse whenever a label-producing mode is active, and name the field. Because `train_on_responses_only` defaults to `true`, refusing would make every default `packing: true` config fail, so wiring is the better option.

The same TRL fallback (`labels = input_ids`) is what #1054 hits for a pre-tokenized cache without labels. A pre-tokenized cache that does carry labels still loses them here once packing is on.

## Acceptance

- [ ] A CPU test built on the `_sft_wrapper` helper from `tests/test_issue691_sft_packing.py` with rows that carry a system and a user prompt: the number of `labels != -100` positions over the train dataloader with `packing: true` equals the count with `packing: false`, give or take the per-sequence boundary token. Currently it is 344 against 64.
- [ ] Variants: `train_on_responses_only: true` (default); `train_on_messages_with_train_field: true`; `training.train_on_eot: true`; a `pre_tokenized` dataset that carries `labels`; `packing_strategy` `bfd`, `bfd-requeue` and `wrapped` (`wrapped` splits sequences, so the mask must be split with them); `padding_free`.
- [ ] `task: pretrain` with `packing: true` is unchanged (no mask involved).
- [ ] If the refuse option is chosen instead: a parse-time error naming `training.packing` and the masking field.
